File: leetcode8.py
class Solution:
    def myAtoi(self, str: str) -> int:
        flag = 1
        result = 0
        id = 0
        str = str.strip()
        for a in str:
            if ord(a) >= ord('0') and ord(a) <= ord('9'):
                result = result * 10 + ord(a) - ord('0')
            elif a == '-' and 0 == id:
                flag = -1*flag
            elif a == '+' and 0 == id:
                pass
            else:
                break
            id += 1
        result = min(result*flag,2**31-1) if result * flag > 0 else max(result*flag,-2**31) 
        return result

a = Solution()
print(a.myAtoi("   -42"))

File: leetcode8.go
// +build ignore

package main
import "fmt"

import "strings"
import "math"

func myAtoi(str string) int {
	str = strings.Trim(str, " ")
	n := len(str)
	result := 0
	flag := 1
	for i := 0; i<n; i++{
		c := str[i]
		if int(c) >= int('0') && int(c) <= int('9'){
			var bignum int64 = (int64(result) * 10 + int64(c) - int64('0'))*int64(flag)
			if bignum > math.MaxInt32{
				return math.MaxInt32
			} else if bignum < math.MinInt32{
				return math.MinInt32
			} else{
				result = result * 10 + int(c) - int('0')
			}
		} else if c == '-' && 0 == i{
			flag *= -1
		} else if c != '+' || 0 != i{
			break
		}
	}
    return flag*result
}
func main(){
	a := myAtoi("  42")
	fmt.Printf(string(a))
}